### PR TITLE
convention to disallow components at invalid locations

### PR DESCRIPTION
**Pull request by udyansharma-01**

### CODE REVIEW

File: tests/lib/rules/no-invalid-resource-location.js [hunk 1]
```diff
@@ -36,20 +36,26 @@ ruleTesterInstance.run('no-invalid-resource-location', ruleUnderTest, {
...
+        },
       ],
       filename: 'frontend/src/modules/common/components/dialog.component.tsx',
     },
     {
-      code: 'export const useAuth = ()=>{/* definition */}',
+      code: 'export const dialogPartial = ()=>{/* definition */}',
       errors: [
         {
           message: 'All partials and their related styles should be placed either in their own directory (.../partials/{partial-name}/{partial-name}.partial.tsx) or alongwith a related component in their directory',
```
Comment (stasys-rnn) [inline tests/lib/rules/no-invalid-resource-location.js:54]: @udyansharma-01 I'm not sure that this message is correct, partials should always be in separate directory.

File: tests/lib/rules/no-invalid-resource-location.js [hunk 2]
```diff
@@ -36,20 +36,26 @@ ruleTesterInstance.run('no-invalid-resource-location', ruleUnderTest, {
   ],
```
Comment (stasys-rnn) [inline tests/lib/rules/no-invalid-resource-location.js:36]: I think it makes sense to add more positive/negative cases for partials (in layouts, modules components).

**Review by stasys-rnn — CHANGES_REQUESTED** [review]
Please check comments.

File: tests/lib/rules/no-invalid-resource-location.js (hunk 2 above)
Comment (udyansharma-01) [reply tests/lib/rules/no-invalid-resource-location.js:36]: Nice catch!

**Review by stasys-rnn — APPROVED** [review]

**Pull request merged**